core/state/state_prove.go

```diff
@@ -0,0 +1,86 @@
+package state
+
+import (
+	"errors"
+	"fmt"
+
+	zkt "github.com/scroll-tech/zktrie/types"
+
+	zktrie "github.com/scroll-tech/go-ethereum/trie"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/crypto"
+	"github.com/scroll-tech/go-ethereum/ethdb"
+)
+
+type TrieProve interface {
+	Prove(key []byte, fromLevel uint, proofDb ethdb.KeyValueWriter) error
+}
+
+type ZktrieProofTracer struct {
+	*zktrie.ProofTracer
+}
+
+// MarkDeletion overwrite the underlayer method with secure key
+func (t ZktrieProofTracer) MarkDeletion(key common.Hash) {
+	key_s, _ := zkt.ToSecureKeyBytes(key.Bytes())
+	t.ProofTracer.MarkDeletion(key_s.Bytes())
+}
```

Comment:
Two reasons:

1. Help to cover the different trie option (eth trie or zktrie) from `tracers` module.
2. Help to keep some entries being consistent: The `Prove` method in trie always handle keys directly (i.e. without the additional secured process being applied on the key), and we expect the 'MarkDeletion' also respect this promise. Since a key is always proven first, then being marked as deletion. And `stateDb` module has the response to pass a 'secured' key to `Prove` (also `MarkDeletion): https://github.com/scroll-tech/go-ethereum/blob/ddaab71dbb8a4ccb6ba5b5888d18bf28e763a3b5/core/state/statedb.go#L331